server/utils/MailService.js

```diff
@@ -0,0 +1,94 @@
+import nodemailer from 'nodemailer';
+import path from 'path';
+import EventEmitter from 'events';
+import isString from 'lodash/isString';
+import isPlainObject from 'lodash/isPlainObject';
+import typeOf from 'typeof';
+
+class MailService extends EventEmitter {
+  constructor(props) {
+    super();
+
+    const {
+      transport = 'debug',
+      from = 'admin@yeep.com',
+      templatePath = path.resolve(__dirname, '../views/'),
+      options = {},
+    } = props;
+    const { auth, service, host } = options;
+
+    if (transport === 'debug') {
```

Comment:
How about we use a `switch` statement instead of an if. The default should be to throw an error.

```
  switch (props.transport) {
    case 'sendgrid':
      // do something
      break;
    default:
      throw new TypeError('Invalid "trasport" property')
  }
```

Reply:
as before. Currently if not provided the transport will always be debug so it shouldn't really throw an error. I could change this into switch statements, sure

Reply:
Also, linter is erroring when i try to assign a variable to a case class. So what do we do about the switch statements?

Reply:
Correct, here's how you can make the linter happy...

```javascript
switch (props.transport) {
  case 'sendgrid': {
    // do something
    // this is now a block scope so feel free to use const and let
    break;
  }
  default:
    throw new TypeError('Invalid "trasport" property')
}
```